gcp/app/server.go

```
@@ -117,3 +145,132 @@ func getExtendedAt(instance *compute.Instance) (time.Time, error) {
 	}
 	return time.Parse(time.RFC3339, instance.CreationTimestamp)
 }
+
+func getShutdownAt(instance *compute.Instance) (time.Time, error) {
+	for _, metadata := range instance.Metadata.Items {
+		if metadata.Key == gcp.MetadataKeyShutdownAt {
+			return time.Parse(time.RFC3339, *metadata.Value)
+		}
+	}
+	return time.Time{}, errShutdownMetadataNotFound
+}
+
+// HandleHandle handles REST API /extend
+func (s Server) HandleExtend(w http.ResponseWriter, r *http.Request) {
+	s.extend(w, r)
+}
+
+func (s Server) extend(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	bodyRaw, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Error("failed to read body", map[string]interface{}{
+			log.FnError: err,
+		})
+		RenderError(r.Context(), w, InternalServerError(err))
+		return
+	}
+
+	body, _ := url.QueryUnescape(string(bodyRaw))
```

Is it OK to ignore this err?